libbuteosyncfw/common/NetworkManager.cpp

```diff
@@ -110,12 +142,54 @@ void NetworkManager::sessionConnectionTimeout()
 {
     if (!m_errorEmitted && m_networkSession) {
         if (!m_networkSession->isOpen()) {
-            qWarning() << "No network reply received after 10 seconds, emitting session error.";
+            LOG_WARNING("No network reply received after 10 seconds, emitting session error.");
             slotSessionError(m_networkSession->error());
         }
     }
 }
 
+void NetworkManager::slotConfigurationChanged()
+{
+    // wait for 3 secs before update connection status
+    // this avoid problems with connections that take a while to be stabilished
+    m_idleRefreshTimer.start(3000);
+}
+
+void NetworkManager::idleRefresh()
+{
+    FUNCTION_CALL_TRACE;
+    QList<QNetworkConfiguration> activeConfigs = m_networkConfigManager->allConfigurations(QNetworkConfiguration::Active);
+    QNetworkConfiguration::BearerType connectionType = QNetworkConfiguration::BearerUnknown;
+    QString bearerTypeName;
+
+    bool isOnline = activeConfigs.size() > 0;
+    if (isOnline)
+    {
+        // FIXME: due this bug #1444162 on nm the QNetworkConfigurationManager
+        // returns the wrong default connection.
+        // We will consider the connection with the smallest bearType as the
+        // default connection, with that wifi and ethernet will be the first one
+        connectionType = activeConfigs.first().bearerType();
+        bearerTypeName = activeConfigs.first().bearerTypeName();
+        foreach(const QNetworkConfiguration &conf, activeConfigs)
+        {
+            if (conf.bearerType() < connectionType)
+            {
+                connectionType = conf.bearerType();
+                bearerTypeName = conf.bearerTypeName();
+            }
+        }
+    }
+    LOG_DEBUG("New state:" << isOnline << " New type: " << bearerTypeName << "(" << connectionType << ")");
+    if ((isOnline != m_isOnline) ||
+        (Sync::InternetConnectionType(connectionType) != m_connectionType))
+    {
+        m_isOnline = isOnline;
+        m_connectionType =  Sync::InternetConnectionType(connectionType);
+        emit statusChanged(m_isOnline, m_connectionType);
```

Comment:
the old slotOnlineStateChanged() used to emit valueChanged() instead of statusChanged() - is this an intentional change?